### ISSUE TITLE
Possible bug in CRF_ATFloss

### ISSUE THREAD

**Issue by oxrider** [gsig/temporal-fields#5]
I was trying to train with batchSize=32 due to limited GPU memory. The program ran on an exception of `dimension out of range` on line 12 in `CRF_ATFloss.lua`:

>    local x_max = torch.max(x,dim)

Here is the complete traceback:
```
tfields_rgb | Test: [1][3736/3738]    Time 1.045  Data 0.000  top1 100.000 ( 95.171)  top5 100.000 ( 82.818)	
tfields_rgb | Test: [1][3737/3738]    Time 1.042  Data 0.000  top1 100.000 ( 95.172)  top5 100.000 ( 82.822)	
/home/tengyu/torch/install/bin/luajit: bad argument #3 to '?' (dimension out of range at /home/tengyu/torch/extra/cutorch/lib/THC/generated/../THCTensorMathReduce.cuh:636)
stack traceback:
	[C]: at 0x7f189d3f66b0
	[C]: in function 'max'
	./layers/CRF_ATFloss.lua:12: in function 'slse'
	./layers/CRF_ATFloss.lua:195: in function 'forward'
	./crftrain.lua:175: in function 'test'
	main.lua:66: in main chunk
	[C]: in function 'dofile'
	exp/tfields_rgb.lua:43: in main chunk
	[C]: in function 'dofile'
	...ngyu/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: at 0x00406670
```

It looks like the error occurred during the last testing batch of epoch 1 when calculating loss.

**Comment by gsig** [comment]
My guess is that batch 3738 had exactly 1 example, so when CRF_ATFloss calls [squeeze()](https://github.com/gsig/temporal-fields/blob/33b58df6ffb6c319ea764ff73c9eef96cbe5b66e/layers/CRF_ATFloss.lua#L194) the tensor shrinks down by an additional dimension.

I will push a commit to the dataloader to always have full batches. In the meantime you can change your call script to have slightly different test size ('-testSize','0.095',)

**Comment by gsig** [comment]
Should be fixed in [53c7715](https://github.com/gsig/temporal-fields/commit/53c77159b4d847f056557c0e687b33368f099898)

Let me know if this keeps happening.

**Comment by oxrider** [comment]
Thank you very much. I'm running with the new dataloader now and will let you know if this keeps happening.
